### ISSUE TITLE
SA1118 should permit `with` use where object initializers are allowed

### ISSUE THREAD

**Issue by AArnott** [DotNetAnalyzers/StyleCopAnalyzers#3314]
This syntax does not produce any warnings:

```cs
this.generator = new Generator(
    this.metadataStream,
    options: new GeneratorOptions
    {
        EmitSingleFile = true,
        WideCharOnly = false,
        ComInterop = new GeneratorOptions.ComInteropOptions { StructsInsteadOfInterfaces = comStructs },
    },
    compilation: this.compilation,
    parseOptions: this.parseOptions);
Assert.True(this.generator.TryGenerate(api, CancellationToken.None));
```

But the following syntax produces SA1118: The parameter spans multiple lines

```cs
this.generator = new Generator(
    this.metadataStream,
    options: DefaultTestGeneratorOptions with
    {
        WideCharOnly = false,
        ComInterop = new GeneratorOptions.ComInteropOptions { StructsInsteadOfInterfaces = comStructs },
    },
    compilation: this.compilation,
    parseOptions: this.parseOptions);
Assert.True(this.generator.TryGenerate(api, CancellationToken.None));
this.CollectGeneratedCode(this.generator);
this.AssertNoDiagnostics();
```
